src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -491,10 +518,23 @@ std::shared_ptr<BinaryTreeMachineNode<CARTreeNodeData>> CARTree::CARTtrain(std::
 	right_child->data.transit_into_values=right_transit;
 	node->data.num_leaves=left_child->data.num_leaves+right_child->data.num_leaves;
 	node->data.weight_minus_branch=left_child->data.weight_minus_branch+right_child->data.weight_minus_branch;
-
+	node->data.impurity = node_impurity;
 	return node;
 }
-
+SGVector<float64_t> CARTree::get_feature_importance()
+{
+	require(m_num_feat, "numbers of features should be larger than 0");
+	m_feature_importances = SGVector<float64_t>(m_num_feat);
+	compute_feature_importance(m_root->as<bnode_t>());
+	float64_t total_num_sample = m_root->data.total_weight;
+	linalg::scale(
+	    m_feature_importances, m_feature_importances, 1 / total_num_sample);
+	auto normalizer = linalg::sum(m_feature_importances);
+	if (normalizer > 0)
+		linalg::scale(
+		    m_feature_importances, m_feature_importances, 1 / normalizer);
+	return m_feature_importances;
```

Comment:
did you compare the feature importance calculation with another implementation?

Reply:
is this the same as in scikit learn?

Reply:
yes, this implementation is the same as sklearn , actually, I borrow it from [this](https://github.com/scikit-learn/scikit-learn/blob/0abd95f742efea826df82458458fcbc0f9dafcb2/sklearn/tree/_tree.pyx#L1056)

Reply:
cool, thanks! seems to be correct :) maybe you could also add a unittest? not sure if it comparable to a result in scikit learn?

Reply:
also you should add a comment in the documentation saying that this is the implementation from scikit learn, as different tree implementations might slightly different definitions of feature importance

Reply:
ok